docs/user_guide/building_blocks/config_objects.md

```diff
@@ -53,9 +125,374 @@ given via the ```--variables``` argument, accessing information from columns bas
+# (used during plotting)
+cfg.x.category_groups = {}
+
+# variable groups for conveniently looping over certain variables
+# (used during plotting)
+cfg.x.variable_groups = {}
+
+# shift groups for conveniently looping over certain shifts
+# (used during plotting)
+cfg.x.shift_groups = {}
+
+# general_settings groups for conveniently looping over different values for the general-settings parameter
+# (used during plotting)
+cfg.x.general_settings_groups = {}
+
+# process_settings groups for conveniently looping over different values for the process-settings parameter
+# (used during plotting)
+cfg.x.process_settings_groups = {}
+
+# variable_settings groups for conveniently looping over different values for the variable-settings parameter
+# (used during plotting)
+cfg.x.variable_settings_groups = {}
+
+# custom_style_config groups for conveniently looping over certain style configs
+# (used during plotting)
+cfg.x.custom_style_config_groups = {}
+
+# calibrator groups for conveniently looping over certain calibrators
+# (used during calibration)
+cfg.x.calibrator_groups = {}
+
+# producer groups for conveniently looping over certain producers
+# (used during the ProduceColumns task)
+cfg.x.producer_groups = {}
+
+# ml_model groups for conveniently looping over certain ml_models
+# (used during the machine learning tasks)
+cfg.x.ml_model_groups = {}
+```
+
+### Reduced Files size
+
+The target size of the files in MB after the {py:class}`~columnflow.tasks.reduction.MergeReducedEvents` task can be set in the Config object with the `cfg.x.reduced_file_size` argument. A float number corresponding to the size in MB is expected. If nothing is set, the default value implemented in columnflow will be used (512MB at the time of writing). An example is given below.
+
+```python
+# target file size after MergeReducedEvents in MB
+cfg.x.reduced_file_size = 512.0
+```
+
+
+### Object-specific variables
+
+Other than the variables mentioned above, several might be needed for specific Producers for example, but these won't be discussed here as they are not general parameters.
+Hence, we invite the users to check which Config entries are needed for each {py:class}`~columnflow.calibration.Calibrator`,
+{py:class}`~columnflow.selection.Selector` and {py:class}`~columnflow.production.Producer` and in general each CMS-specific object (=objects in the cms-subfolders) they want to use.
+Since the {py:class}`~columnflow.production.cms.muon.muon_weights` Producer was already mentioned above, we will remind users here that the `cfg.x.muon_sf_names` Config entry is needed for this Producer to run, as indicated in the docstring of the Producer.
+
+As for the CMS-specific objects, an example could be the task {py:class}`~columnflow.tasks.cms.external.CreatePileupWeights`, which requires additional entries in the Config objects, for example the minimum bias cross sections `cfg.x.minbias_xs` and the `pu` entry to the external files.
```

Comment:
You already made it clear that this list here requires additional information from the config

```suggestion
As for the CMS-specific objects, an example could be the task {py:class}`~columnflow.tasks.cms.external.CreatePileupWeights`, which requires for example the minimum bias cross sections `cfg.x.minbias_xs` and the `pu` entry in the external files.
```